core/src/main/java/io/aiven/klaw/service/ClusterApiService.java

```diff
@@ -727,6 +717,75 @@ public ApiResponse getAivenServiceAccounts(String projectName, String serviceNam
     }
   }
 
+  ResponseEntity<ApiResponse> deleteSchema(String topicName, String kafkaEnvId, int tenantId)
+      throws KlawException {
+    log.info("delete schema subject on cluster {} {}", topicName, kafkaEnvId);
+    getClusterApiProperties(tenantId);
+    ResponseEntity<ApiResponse> response;
+    ClusterTopicRequest clusterTopicRequest;
+    try {
+      Env envSelected = manageDatabase.getHandleDbRequests().getEnvDetails(kafkaEnvId, tenantId);
+      KwClusters kwClusters =
+          manageDatabase
+              .getClusters(KafkaClustersType.KAFKA, tenantId)
+              .get(envSelected.getClusterId());
+      clusterTopicRequest =
+          ClusterTopicRequest.builder()
+              .env(kwClusters.getBootstrapServers())
+              .protocol(kwClusters.getProtocol())
+              .clusterName(kwClusters.getClusterName() + kwClusters.getClusterId())
+              .topicName(topicName)
+              .build();
+
+      String uri;
+
+      uri = clusterConnUrl + URI_DELETE_SCHEMAS;
+      if (envSelected.getAssociatedEnv() != null) {
+        // get associated schema env
+        Env schemaEnvSelected =
+            manageDatabase
+                .getHandleDbRequests()
+                .getEnvDetails(envSelected.getAssociatedEnv().getId(), tenantId);
```

Comment:
do we need to check for associated Env being null?

Reply:
there is an if condition already declared